backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/contract/contractPeriod/extendModularContractPeriod.feature

```diff
@@ -0,0 +1,105 @@
+      | y2022                | 2022       | harvesting_calendar          |
+
+
+  @from:cucumber
+  Scenario: Fail Extend - Contract Period with Contract Term of type Modular Contract & OnFlatrateTermExtend = Extension Not Allowed is OK
+  - Contract Conditions of type Modular Contract & OnFlatrateTermExtend is Extension Not Allowed already exists
+  - Contract Period of Modular Contract already exists
+  - extend contract period for a given date
+  - fail with message : Extension Not Allowed
+
+    Given metasfresh contains M_PricingSystems
+      | Identifier               | Name                                | Value                               |
+      | modularCntrPricingSystem | modularCntrPricingSystem_23072023_1 | modularCntrPricingSystem_23072023_1 |
+
+    And metasfresh contains M_Products:
+      | Identifier         | Name                               |
+      | module_log_product | module_log_product_test_07042023_1 |
+
+    And metasfresh contains C_BPartners:
+      | Identifier           | Name                            | OPT.IsVendor | M_PricingSystem_ID.Identifier | OPT.C_PaymentTerm_ID.Value |
+      | bp_modularCntrPeriod | bp_modularCntrPeriod_23072023_1 | Y            | modularCntrPricingSystem      | 1000002                    |
+
+    And metasfresh contains C_BPartner_Locations:
+      | Identifier                    | GLN           | C_BPartner_ID.Identifier | OPT.IsShipToDefault | OPT.IsBillToDefault |
+      | bp_modularCntrPeriod_Location | 5823198505483 | bp_modularCntrPeriod     | true                | true                |
+
+    And metasfresh contains ModCntr_Settings:
+      | ModCntr_Settings_ID.Identifier | Name                | M_Product_ID.Identifier | C_Calendar_ID.Identifier | C_Year_ID.Identifier | OPT.M_PricingSystem_ID.Identifier |
+      | modCntr_settings_1             | settings_23072023_1 | module_log_product      | harvesting_calendar      | y2022                | modularCntrPricingSystem          |
+
+    And metasfresh contains C_Flatrate_Conditions:
+      | C_Flatrate_Conditions_ID.Identifier | Name                             | Type_Conditions | OPT.M_PricingSystem_ID.Identifier | OPT.OnFlatrateTermExtend | OPT.ModCntr_Settings_ID.Identifier |
+      | modularCntrConditions_1             | modularCntrConditions_23072023_1 | ModularContract | modularCntrPricingSystem          | Ex                       | modCntr_settings_1                 |
+
+    And metasfresh contains C_Flatrate_Terms:
+      | Identifier        | C_Flatrate_Conditions_ID.Identifier | Bill_BPartner_ID.Identifier | StartDate  | EndDate    | OPT.M_Product_ID.Identifier |
+      | modularCntrTerm_1 | modularCntrConditions_1             | bp_modularCntrPeriod        | 2021-10-31 | 2022-10-30 | module_log_product          |
+
+    When extend C_Flatrate_Term identified by modularCntrTerm_1 starting from 2022-10-31 will fail with message Verlängerung nicht zulässig
```

Comment:
```suggestion
    Then extend C_Flatrate_Term identified by modularCntrTerm_1 starting from 2022-10-31 will fail with message "Verlängerung nicht zulässig"
```

as suggested above.
Also, please try to write the scenarios in the Given - When - Then format